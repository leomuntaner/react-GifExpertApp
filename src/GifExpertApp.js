import React,{ useState } from 'react'
import { AddCategory} from './components/AddCategory';
import PropTypes from 'prop-types';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch']);

    return (
     <>        
        <h2>GifExpertApp</h2>
        <AddCategory setCategories={ setCategories }/>
        <hr/> 
        <ol>
            {
                categories.map( category =>{
                  return <GifGrid 
                            category={ category }
                            key= { category }    
                         />
                })
            }
        </ol>

     </>
    )

};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};
