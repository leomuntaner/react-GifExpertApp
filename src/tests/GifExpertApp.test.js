import React from 'react';
import {GifExpertApp} from '../GifExpertApp';
import {shallow} from 'enzyme';

describe(' Tests en Gif Expert App ', () => {
    
    
    test('Debe mostrarse crrectamente', () => {
        
        const wrapper = shallow( <GifExpertApp   />);

        expect( wrapper ).toMatchSnapshot();
    })
    
    test('DEBE DE MOSTRAR UNA LISTA DE CATEGORIAS', () => {
        
        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow( <GifExpertApp defaultCategories={categories}   />);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe(categories.length);
    })
    

})