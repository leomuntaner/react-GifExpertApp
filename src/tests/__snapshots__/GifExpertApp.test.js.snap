// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Tests en Gif Expert App  DEBE DE MOSTRAR UNA LISTA DE CATEGORIAS 1`] = `ShallowWrapper {}`;

exports[` Tests en Gif Expert App  Debe mostrarse crrectamente 1`] = `ShallowWrapper {}`;
